agent/engine/engine_integ_test.go

```diff
@@ -831,3 +833,74 @@ func TestPullContainerWithAndWithoutDigestConsistency(t *testing.T) {
 	// Image should be the same
 	assert.Equal(t, inspectWithDigest.ID, inspectWithoutDigest.ID)
 }
+
+// Tests that a task with invalid image fails as expected.
+func TestInvalidImageInteg(t *testing.T) {
+	// Prepare task engine
+	cfg := defaultTestConfigIntegTest()
+	cfg.ImagePullBehavior = config.ImagePullAlwaysBehavior
+	taskEngine, done, _ := setup(cfg, nil, t)
+	defer done()
+
+	// Prepare a task
+	container := createTestContainerWithImageAndName("127.0.0.1:51670/invalid-image", "container")
+	task := &apitask.Task{
+		Arn:                 "test-arn",
+		Family:              "family",
+		Version:             "1",
+		DesiredStatusUnsafe: apitaskstatus.TaskRunning,
+		Containers:          []*apicontainer.Container{container},
+	}
+
+	// Start the task
+	go taskEngine.AddTask(task)
+
+	// The container and the task both should stop
+	verifyContainerStoppedStateChangeWithReason(t, taskEngine,
+		"CannotPullImageManifestError: Error response from daemon: manifest unknown: manifest unknown")
+	verifyTaskStoppedStateChange(t, taskEngine)
+}
+
+// Tests that a task with an image that has a digest specified works normally.
+func TestImageWithDigestInteg(t *testing.T) {
+	// Prepare task engine
+	cfg := defaultTestConfigIntegTest()
+	cfg.ImagePullBehavior = config.ImagePullAlwaysBehavior
+	taskEngine, done, _ := setup(cfg, nil, t)
+	defer done()
+
+	// Find image digest
+	versionedClient, err := taskEngine.(*DockerTaskEngine).client.WithVersion(dockerclient.Version_1_35)
+	manifest, manifestPullError := versionedClient.PullImageManifest(
+		context.Background(), localRegistryBusyboxImage, nil)
+	require.NoError(t, manifestPullError)
+	imageDigest := manifest.Descriptor.Digest.String()
+
+	// Prepare a task with image digest
+	container := createTestContainerWithImageAndName(
+		localRegistryBusyboxImage+"@"+imageDigest, "container")
```

Comment:
Is there a case to be tested for invalid digest?

Reply:
That should behave the same as `TestInvalidImageInteg` but I will add it nevertheless

Reply:
Added the test case